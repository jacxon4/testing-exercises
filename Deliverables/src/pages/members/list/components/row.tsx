import * as React from 'react';
import { Member } from '../viewModel';

export interface RowProps {
  member: Member;
}

export const Row: React.StatelessComponent<RowProps> = (props) => (
  <tr>
    <td>
      <img src={props.member.avatarUrl} style={{ maxWidth: '10rem' }} />
    </td>
    <td>
      <span>{props.member.id}</span>
    </td>
    <td>
      <span>{props.member.name}</span>
    </td>
  </tr>
);
