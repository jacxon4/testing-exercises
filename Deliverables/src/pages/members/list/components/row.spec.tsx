import * as React from 'react';
import { shallow } from 'enzyme';
import { Row, RowProps } from './row';

describe('pages/members/list/components/row', () => {
  it('should create a tr with given props', () => {
    const props: RowProps = {
      member: { avatarUrl: 'avatarURL1', name: 'name1', id: 1 },
    };

    const component = shallow(<Row {...props} />);

    expect(component).toMatchSnapshot();
  });
});
