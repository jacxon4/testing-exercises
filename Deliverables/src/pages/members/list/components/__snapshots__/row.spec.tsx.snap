// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/members/list/components/row should create a tr with given props 1`] = `
<tr>
  <td>
    <img
      src="avatarURL1"
      style={
        Object {
          "maxWidth": "10rem",
        }
      }
    />
  </td>
  <td>
    <span>
      1
    </span>
  </td>
  <td>
    <span>
      name1
    </span>
  </td>
</tr>
`;
